### PR TITLE
🧬 Update and fix all issues of Udemy, Medium and Dizilla

### LINKED ISSUE #3444: [Presence | Bug] Medium
# Describe the Bug
Currently, the Medium presence is partially broken. I have noticed many issues with it, for example it always returns "Viewing the page: Home" when it couldn't retrieve the information from the page or displays the "Reading a story" presence when it's not needed.

# Steps to Reproduce

### First issue - Authors are never displayed
1. Install the Medium presence from the store
2. Go to: https://medium.com/ and click to any story
3. Watch your presence change with the correct title but "Open in app" as author

### Second issue - New stories are displayed as "Home"
1. Install the Medium presence from the store
2. Login to your Medium account and go to: https://medium.com/new-story
3. Watch your presence change to "Writing a new story: No title set"
4. Start typing a title and press enter
5. Observe your presence turning into "Viewing the page: Home"

### Third issue - Viewing the list of stories turns into "Reading a story"
1. Install the Medium presence from the store
2. Login to your Medium account and go to: https://medium.com/me/stories/public
3. Possibly have at least one story to make it happen
4. Watch your presence weirdly turn into "Reading a story" presence

# Expected Behavior
### First issue - Authors are never displayed
The author's name is supposed to be displayed (Sean Kernan for my example in my screenshot below)

### Second issue - New stories are displayed as "Home"
The initial presence is supposed to be updated, "Writing a new story: No title set" should now get the title once you start typing (the first paragraph is always considered as title by Medium, doesn't matters if you follow the title preset of Medium or not).

### Third issue - Viewing the list of stories turns into "Reading a story"
The presence should turn into "Viewing a page: Your stories"

# Screenshots
presence
author-not-displayed
list-of-stories-issue

# Environment
- OS: Windows 10 Version 10.0.19042 Build 19042
- Browser: Chrome
- Version: 89.0.4389.114 (Official build) (64 bits)

# Additional context

- [x] Checked if the PreMiD extension and client are up to date
- [x] Checked through (closed and open) issues to see if the issues above were reported before

### LINKED ISSUE #3537: [Presence | Bug] Udemy
**Describe the Bug**
The presence doesn't recognize the Course I'm watching

**Steps to Reproduce**
Watch any Course in Udemy website
Verify if the presence is working as it should

**Expected Behavior**
The presence should show the name of the Course and the episode, instead it shows as "Unknown Course" and "Unknown Episode". It doesn't matter what course I'm watching.

**Screenshots**
DiscordCanary_TkannJ9owA

**Environment**

- OS: Windows 10
- Browser: Opera GX / Brave
- Version: Opera GX: LVL2 (core: 73.0.3856.431) / Brave: Versão 1.22.72 Chromium: 89.0.4389.128 (Versão oficial) 64 bits

### PR DESCRIPTION
**Pull request by eggsy**
### Udemy

Fixed all issues and improve overall code (fixes #3537).

image

### Medium

Fixed all (possible) issues and improved the code (fixes #3444).

image

### Dizilla

It's working now.

image

### CODE REVIEW

File: websites/D/Dizilla/presence.ts [hunk 1]
```diff
@@ -113,20 +100,20 @@ presence.on("UpdateData", async () => {
...
 
-    presence.setActivity(object);
+    dizilla.setActivity(object);
   } else if (
     !isNaN(video?.duration) &&
     showName?.innerText &&
     episode?.textContent
   ) {
-    const timestamps = getTimestamps(
+    const timestamps = dizilla.getTimestamps(
```
Comment (ririxidev) [inline websites/D/Dizilla/presence.ts:116]: use array deconstructing

**Review by rxri — CHANGES_REQUESTED** [review]
first;
all of `object.startTimestamp` move to `object` object and value should be `~~(Date.now() / 1000)` or `Math.floor(Date.now() / 1000)`
second;
don't set startTimestamp in videos.

**Review by rxri — APPROVED** [review]

**Review by Slowlife01 — APPROVED** [review]

**Comment by AlperSal** [comment]
Now, on Udemy presence at least, lesson name shows up but course name is still "Unknown Course". Photo;
image

**Comment by AlperSal** [comment]
> 
> 
> Now, on Udemy presence at least, lesson name shows up but course name is still "Unknown Course". Photo;
> image

Any fix?

**Pull request merged**